### DISCUSSION TITLE
8048S043 capacitative touch buttons not working in demo

### DISCUSSION THREAD

**Discussion by ralphrmartin** [rzeldent/esp32-smartdisplay#189] [Q&A]
I have a problem on an 8048s043 board, bought as having capacitative touch. It has both a GT911 chip and an xpt2046 chip.
The board as supplied had a demo running on it iwhich iworked OK, including touch (which does indeed seem to be capacitative as it responded to a light touch), but when I try the eap32-smartdisplay-demo program, touch does not work correctly (whether I select 8048S043C or 8048S043R). The display is drawn correctly, but the buttons to add 1 and to rotate do not function.

On enabling debugging, the display" jumps" whenever I press either button, but they still do not work as intended. The debugging information shows the below. 

Can anyone enlighten me as to what is going wrong here?
(Why is the SDK version: v4.4.7-dirty?)

Thanks.

```

ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x1 (POWERON),boot:0x8 (SPI_FAST_FLASH_BOOT)
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3808,len:0x4bc
load:0x403c9700,len:0xbd8
load:0x403cc700,len:0x2a0c
entry 0x403c98d0
[   139][I][esp32-hal-psram.c:96] psramInit(): PSRAM enabled
[  5149][I][main.cpp:29] setup(): Board: esp32-8048S043C
[  5155][I][main.cpp:30] setup(): CPU: ESP32-S3 rev0, CPU Freq: 240 Mhz, 2 core(s)
[  5162][I][main.cpp:31] setup(): Free heap: 316116 bytes
[  5167][I][main.cpp:32] setup(): Free PSRAM: 8386279 bytes
[  5173][I][main.cpp:33] setup(): SDK version: v4.4.7-dirty
[  5178][D][esp32_smartdisplay.c:173] smartdisplay_init(): smartdisplay_init
[  5185][D][lvgl_panel_st7262_par.c:65] lvgl_lcd_init(): rgb_panel_config: clk_src:0, timings:{pclk_hz:12500000, h_res:800, v_res:480, hsync_pulse_width:4, hsync_back_porch:8, hsync_front_porch:8, vsync_pulse_width:4, vsync_back_porch:8, vsync_front_porch:8, flags:{hsync_idle_low:0, vsync_idle_low:0, de_idle_high:0, pclk_active_neg:1, pclk_idle_high:0}}, data_width:16, sram_trans_align:4, psram_trans_align:64, hsync_gpio_num:39, vsync_gpio_num:41, de_gpio_num:40, pclk_gpio_num:42, data_gpio_nums:[15,16,4,45,48,47,21,14,8,3,46,9,1,5,6,7,], disp_gpio_num:-1, on_frame_trans_done:0x42029c64, user_ctx:0x3fca4718, flags:{disp_active_low:0, relax_on_idle:0, fb_in_psram:1}
[  5244][D][lvgl_panel_st7262_par.c:66] lvgl_lcd_init(): refresh rate: 0 Hz
[  5275][D][lvgl_touch_gt911_i2c.c:45] lvgl_touch_init(): i2c_config: mode:1, sda_io_num:19, scl_io_num:20, sda_pullup_en:0, scl_pullup_en:0, master:{clk_speed:400000}, clk_flags:0
[  5292][D][lvgl_touch_gt911_i2c.c:60] lvgl_touch_init(): io_i2c_config: dev_addr:0x5d, control_phase_bytes:1, user_ctx:0x3fca46f8, dc_bit_offset:0, lcd_cmd_bits:16, lcd_param_bits:0, flags:{.dc_low_on_data:0, disable_control_phase:1}
[  5312][D][lvgl_touch_gt911_i2c.c:75] lvgl_touch_init(): touch_config: x_max:800, y_max:480, rst_gpio_num:38, int_gpio_num:18, levels:{reset:0, interrupt:0}, flags:{swap_xy:0, mirror_x:0, mirror_y:0}, user_data:0x3fcec7d4
[  5384][D][esp_touch_gt911.c:147] gt911_read_info(): GT911 productId: 911
[  5390][D][esp_touch_gt911.c:148] gt911_read_info(): GT911 fwId: 0x1060
[  5397][D][esp_touch_gt911.c:149] gt911_read_info(): GT911 xResolution/yResolution: (0,0)
[  5405][D][esp_touch_gt911.c:150] gt911_read_info(): GT911 vendorId: 0xff
[  5412][D][esp_touch_gt911.c:465] esp_lcd_touch_new_i2c_gt911(): handle:0x3d977b08
[ 35956][D][esp_touch_gt911.c:266] gt911_read_data(): Point: #0, event:0, point:(421,45), area:53
[ 35965][D][esp_touch_gt911.c:303] gt911_get_xy(): Touch data: x:421, y:45, strength:53
[ 35986][D][esp_touch_gt911.c:266] gt911_read_data(): Point: #0, event:0, point:(421,45), area:53
[ 35995][D][esp_touch_gt911.c:303] gt911_get_xy(): Touch data: x:421, y:45, strength:53
[ 36016][D][esp_touch_gt911.c:266] gt911_read_data(): Point: #0, event:0, point:(421,45), area:53
[ 36025][D][esp_touch_gt911.c:303] gt911_get_xy(): Touch data: x:421, y:45, strength:53
[ 36046][D][esp_touch_gt911.c:266] gt911_read_data(): Point: #0, event:0, point:(421,45), area:53
[ 36055][D][esp_touch_gt911.c:303] gt911_get_xy(): Touch data: x:421, y:45, strength:53
[ 36076][D][esp_touch_gt911.c:266] gt911_read_data(): Point: #0, event:0, point:(421,45), area:53
[ 36085][D][esp_touch_gt911.c:303] gt911_get_xy(): Touch data: x:421, y:45, strength:53
[ 36106][D][esp_touch_gt911.c:266] gt911_read_data(): Point: #0, event:0, point:(421,45), area:53
[ 36115][D][esp_touch_gt911.c:303] gt911_get_xy(): Touch data: x:421, y:45, strength:53
[ 36646][D][esp_touch_gt911.c:266] gt911_read_data(): Point: #0, event:0, point:(413,52), area:70
[ 36655][D][esp_touch_gt911.c:303] gt911_get_xy(): Touch data: x:413, y:52, strength:70
[ 36676][D][esp_touch_gt911.c:266] gt911_read_data(): Point: #0, event:0, point:(413,52), area:70
[ 36685][D][esp_touch_gt911.c:303] gt911_get_xy(): Touch data: x:413, y:52, strength:70
[ 36706][D][esp_touch_gt911.c:266] gt911_read_data(): Point: #0, event:0, point:(413,52), area:70
[ 36715][D][esp_touch_gt911.c:303] gt911_get_xy(): Touch data: x:413, y:52, strength:70
[ 36736][D][esp_touch_gt911.c:266] gt911_read_data(): Point: #0, event:0, point:(413,52), area:70
[ 36745][D][esp_touch_gt911.c:303] gt911_get_xy(): Touch data: x:413, y:52, strength:70
[ 36766][D][esp_touch_gt911.c:266] gt911_read_data(): Point: #0, event:0, point:(413,52), area:70
[ 36775][D][esp_touch_gt911.c:303] gt911_get_xy(): Touch data: x:413, y:52, strength:70
[ 36796][D][esp_touch_gt911.c:266] gt911_read_data(): Point: #0, event:0, point:(413,52), area:70
[ 36805][D][esp_touch_gt911.c:303] gt911_get_xy(): Touch data: x:413, y:52, strength:70
[ 36826][D][esp_touch_gt911.c:266] gt911_read_data(): Point: #0, event:0, point:(413,52), area:70
[ 36835][D][esp_touch_gt911.c:303] gt911_get_xy(): Touch data: x:413, y:52, strength:70
[ 36856][D][esp_touch_gt911.c:266] gt911_read_data(): Point: #0, event:0, point:(413,52), area:64
[ 36865][D][esp_touch_gt911.c:303] gt911_get_xy(): Touch data: x:413, y:52, strength:64
```

**Comment by ralphrmartin** [comment]
My basic setup is OK, as I also have a 2432s028 board, and the demo program works perfectly on that, with
`default_envs = esp32-2432S028Rv2`

**Comment by ralphrmartin** [comment]
Reading further, I found that if I changed 
`GT911_TOUCH_CONFIG_INT_GPIO_NUM=18`
to
`GT911_TOUCH_CONFIG_INT_GPIO_NUM=GPIO_NUM_NC`
everything works as expected. I'd suggest this change needs making in the master copy of 
boards/esp32-8048S043C.json

**Comment by rzeldent** [comment]
Hi Ralph,

I retested this with the original setup on the 8048S043C and am not able to reproduce the issue. The interrupt op the GT911 is also not used in the code. Are you sure this is the issue? Anyone else having problems with this board?

**Comment by ralphrmartin** [comment]
Maybe the board design has changed after you bought yours? I see some other boards have variants. Mine was bought very recently. I note that the board came in an antistatic bag with 8048S043C-I on it. I have no idea what the -I stands for. The board itself just says  8048S043 on it. It is definitely a capacitative board. I am 100% sure that the buttons did not work before making the change I noted above, and now it works as expected.

**Comment by ralphrmartin** [comment]
I didn't change the demo in any way, and am using the latest version from git.

**Comment by rzeldent** [comment]
This in my version of the board... Is  your description identical?
![PXL_20240820_202008827 RAW-01 COVER](https://github.com/user-attachments/assets/1262f66c-fb21-4a3d-9053-2c755ec8e3a8)

**Comment by ralphrmartin** [comment]
Yes, same bag, and the board itself looks like this:
![IMG20240821083910](https://github.com/user-attachments/assets/558e3e64-a4f6-434f-96e8-0ce42656697e)

**Comment by Forrest5301** [comment]
My 8048S032C board 800x480 ... akso required martin
GT911_TOUCH_CONFIG_INT_GPIO_NUM=GPIO_NUM_NC
for touch to work ... just received it from AliExpress this week.
Many thanks to ralphrmartin.

**Comment by rzeldent** [comment]
Very strange, mine has exactly the same layout!

**Comment by rzeldent** [comment]
I looked at the code and cannot really find a cause. Only thing that I can think of is the order of initialization.
I created a new branch where this has been changed and also uses the correct interrupt leven pos/neg

This can be found in the branch: https://github.com/rzeldent/esp32-smartdisplay/tree/gt911_int_nc

Can you check if this works for you?

**Comment by ralphrmartin** [comment]
Can you explain how I can use that branch with the smart display demo app? I'm afraid I dont know my way around git enough to do that without some explicit instructions.

**Comment by ralphrmartin** [comment]
OK, I made a new demo project, and changed .gitmodules to say
```
[submodule "boards"]
	path = boards
	url =  https://github.com/rzeldent/esp32-smartdisplay/tree/gt911_int_nc.git
	branch = feature/iodefs
```
and build with `default_envs = esp32-8048S043C` and flashed the project to my board, and in this case the touch again did **NOT** work. I then changed the board json file to say `GT911_TOUCH_CONFIG_INT_GPIO_NUM=GPIO_NUM_NC` and it did work. Maybe I did not make the correct change to .gitmodules  - please let me know if my change was incorrect.

**Comment by rzeldent** [comment]
Hi Ralph,

Yes, that is indeed how you should do this; update the URL. I normally dop this with #gt911_int_nc.
Question: did you clear the .pio directory because this is cached!

**Reply by ralphrmartin** [reply]
I made a completely new demo project and threw the old one away before trying this out, so there was no question of caching in the .pio folder.

**Comment by rzeldent** [comment]
Hi Ralph,

Thanks for testing. Looks like there is something different but still not figured out what.
What is strange is that the interrupts are not used while reading, and also no handler is set.
Needs more investigation...
